app/components/scan-qr/qrcode-scanner.tsx

```diff
@@ -48,13 +48,13 @@ const QRScanner: React.FC<QRScannerProps> = ({ onClose }) => {
 
               if (match) {
                 stopMediaStream();
-                // const qrId = match[4]; // Get the last segment of the URL as the QR id
+                const qrId = match[4]; // Get the last segment of the URL as the QR id
 
                 setScanCompleted(true); // Set the scanCompleted state to true
 
-                window.location.href = scannedData;
+                // window.location.href = scannedData;
 
-                // navigate(`/qr/${qrId}`); (using this way will not close camera access)
+                navigate(`/qr/${qrId}`); (using this way will not close camera access)
```

Comment:
There is an error here. You left part of the comment

Reply:
corrected